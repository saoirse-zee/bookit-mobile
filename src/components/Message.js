import React from 'react'
import { StyleSheet, Text, View, TouchableHighlight } from 'react-native'
import colors from '../../constants/Colors'

const Message = ({ message, onPress, buttonLabel }) => (
  <View style={message ? styles.container : [styles.container, styles.emptyContainer]}>
    <Text style={styles.text}>
      { message }
    </Text>
    {
      onPress ? (
        <TouchableHighlight
          onPress={onPress}
          underlayColor={colors.tintColorDarkened}
        >
          <Text style={styles.text}>{buttonLabel}</Text>
        </TouchableHighlight>
      ) : null
    }
  </View>
)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    maxHeight: 40,
    paddingLeft: 30,
    backgroundColor: colors.tintColor,
  },
  emptyContainer: {
    backgroundColor: undefined,
  },
  text: {
    color: 'white',
    padding: 10,
  },
})

export default Message
