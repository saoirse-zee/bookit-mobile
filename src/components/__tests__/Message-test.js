import 'react-native'
import React from 'react'
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer'
import Message from '../Message'

const stubPressHandler = () => {}

describe('Message box', () => {
  it('displays the given message', () => {
    const tree = renderer.create(<Message
      message="Hey there, friend."
      onPress={stubPressHandler}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has no background color when message prop is missing', () => {
    const tree = renderer.create(<Message
      onPress={stubPressHandler}
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('has no button when onPress prop is missing', () => {
    const tree = renderer.create(<Message
      message="Hey there, friend."
    />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
