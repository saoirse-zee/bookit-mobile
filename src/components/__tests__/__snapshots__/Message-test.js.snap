// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message box displays the given message 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4F97FF",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "maxHeight": 40,
      "paddingLeft": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "padding": 10,
      }
    }
  >
    Hey there, friend.
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Message box has no background color when message prop is missing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#4F97FF",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "maxHeight": 40,
        "paddingLeft": 30,
      },
      Object {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "padding": 10,
      }
    }
  />
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "padding": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Message box has no button when onPress prop is missing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4F97FF",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "maxHeight": 40,
      "paddingLeft": 30,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "padding": 10,
      }
    }
  >
    Hey there, friend.
  </Text>
</View>
`;
